### PR TITLE
enable mypy check for jit_metaprogramming_utils

### LINKED ISSUE #42969: Enable torch.testing._internal typechecks during CI
`TestTypeHints.test_run_mypy_strict` should pass with `[mypy-torch.testing._internal.*]` deleted from `mypy.ini`

cc @ezyang @malfet @rgommers

### PR DESCRIPTION
**Pull request by qxu-fb**
Fixes #42969
enable mypy check for jit_metaprogramming_utils.py and fixed all errors.

### CODE REVIEW

File: torch/testing/_internal/jit_metaprogramming_utils.py [hunk 1]
```diff
@@ -286,7 +288,7 @@ def script_fn(*args, **kwargs):
         fn, tensors = gen_script_fn_and_args(method_name, func_type, *args, **kwargs)
         self.assertExportImport(fn.graph, tensors)
         output = output_process_fn(fn(*tensors))
-        script_fn.last_graph = fn.graph_for(*tensors)
+        setattr(script_fn, 'last_graph', fn.graph_for(*tensors))
```
Comment (walterddr) [inline torch/testing/_internal/jit_metaprogramming_utils.py:291]: I dont think this way of setting function attribute is consider safe under mypy rules. this https://github.com/python/mypy/issues/2087 might be useful

**Comment by qxu-fb** [comment]
Ignore function attributes, as there are yet no good way to type annotate those, see python/mypy#2087

File: torch/testing/_internal/jit_metaprogramming_utils.py [hunk 2]
```diff
@@ -286,7 +288,7 @@ def script_fn(*args, **kwargs):
         fn, tensors = gen_script_fn_and_args(method_name, func_type, *args, **kwargs)
         self.assertExportImport(fn.graph, tensors)
         output = output_process_fn(fn(*tensors))
-        script_fn.last_graph = fn.graph_for(*tensors)
+        script_fn.last_graph = fn.graph_for(*tensors)  # type: ignore[attr-defined]
```
Comment (walterddr) [inline torch/testing/_internal/jit_metaprogramming_utils.py:291]:
```suggestion
        # skip type annotate function attributes for now, see: https://github.com/python/mypy/issues/2087
        script_fn.last_graph = fn.graph_for(*tensors)  # type: ignore[attr-defined]
```

**Review by walterddr — COMMENTED** [review]
looks good to me overall. left one comment.

Once fixed, can you click the "import to Phabricator" button at the bottom of the PR page so that I can approve both?

**Review by walterddr — APPROVED** [review]

**Pull request closed without merging**